### ISSUE TITLE
ipfs-companion clears local storage when using local gateway

### ISSUE THREAD

**Issue by xuhcc** [ipfs/ipfs-companion#977]
**Describe the bug**

If I configure ipfs-companion to use local subdomain gateway it somehow clears local storage every time the page is loaded via gateway, and not only at ipfs.localhost origin, but on all localhost origins. This does not happen when I use remote path-based gateway.

**To Reproduce**
1. Open local page (e.g. http://localhost:1234) and add something to local storage.
2. Open new tab and load something via local IPFS gateway, for example http://blog.ipfs.io.ipns.localhost:8001/
3. Local storage at http://blog.ipfs.io.ipns.localhost:8001/ is cleared
3. Local storage at http://localhost:1234 also cleared

**Expected behavior**

Local storage must not be cleared anywhere.

**Desktop:**

 - OS: Linux
 - Browser: Firefox 84
 - IPFS companion: 2.17.3

**Comment by lidel** [comment]
Thank you for reporting this @xuhcc, nice catch!

I suspect this might be because we use [path-based router which then returns a redirect to the proper subdomain](https://docs.ipfs.io/how-to/address-ipfs-on-web/#automatic-leverage-the-gateway-in-go-ipfs).
It is the same as redirect as when opening  https://dweb.link/ipns/en.wikipedia-on-ipfs.org and ending up at  https://en-wikipedia--on--ipfs-org.ipns.dweb.link

The caveat is that the redirect is returned with `Clear-Site-Data: "cookies", "storage"` HTTP header which was added to go-ipfs as precaution in early days of subdomain gateways.  This header should be ignored by browsers on redirects, but it seems that Firefox acts on it (I confirmed that Brave does not clear cookies). 

I believe we should remove `Clear-Site-Data` from redirect responses: 

- subdomain gateways ended up being self-contained. Once a gateway domain is defined in config, it is impossible to load websites from paths, we always return redirect.
- `Clear-Site-Data` support across vendors is buggy and  that is not just Firefox: Chromium been ignoring a bug for three years: https://bugs.chromium.org/p/chromium/issues/detail?id=898503)

Will prepare upstream fix shortly and will try to include it in go-ipfs 0.8.0. 
Until that happens, a workaround is to use Brave.
